go/cmd/dolt/commands/show.go

```diff
@@ -111,16 +125,109 @@ func (cmd ShowCmd) Exec(ctx context.Context, commandStr string, args []string, d
 		return handleErrAndExit(err)
 	}
 
-	if !opts.pretty && !dEnv.DoltDB.Format().UsesFlatbuffers() {
-		cli.PrintErrln("dolt show --no-pretty is not supported when using old LD_1 storage format.")
-		return 1
+	opts.diffDisplaySettings = parseDiffDisplaySettings(apr)
+
+	// Decide if we're going to use dolt or sql for this execution.
+	// We can use SQL in the following cases:
+	// 1. `--no-pretty` is not set, so we will be producing "pretty" output.
+	// 2. opts.specRefs contains values that are NOT commit hashes.
+	// In all other cases, we should use DoltEnv
+	allSpecRefsAreCommits := true
+	allSpecRefsAreNonCommits := true
+	resolvedNonCommitSpecs := map[string]string{}
+	for _, specRef := range opts.specRefs {
+		isNonCommitSpec, resolvedValue, err := resolveNonCommitSpec(ctx, dEnv, specRef)
+		if err != nil {
+			err = fmt.Errorf("error resolving spec ref '%s': %w", specRef, err)
+			return handleErrAndExit(err)
+		}
+		allSpecRefsAreNonCommits = allSpecRefsAreNonCommits && isNonCommitSpec
+		allSpecRefsAreCommits = allSpecRefsAreCommits && !isNonCommitSpec
+
+		if isNonCommitSpec {
+			resolvedNonCommitSpecs[specRef] = resolvedValue
+		}
+	}
+
+	if !allSpecRefsAreCommits && !allSpecRefsAreNonCommits {
+		err = fmt.Errorf("cannot mix commit hashes and non-commit spec refs")
+		return handleErrAndExit(err)
+	}
+
+	useDoltEnv := !opts.pretty || (len(opts.specRefs) > 0 && allSpecRefsAreNonCommits)
+
+	if useDoltEnv {
+		if !opts.pretty && !dEnv.DoltDB.Format().UsesFlatbuffers() {
```

Comment:
I think it would be a good idea to at least determine if a server is running, then print a warning or possible abort if so. We don't want our users to be surprised by the fact that this is not a supported code path for remote calls.

Reply:
Testing for this now.

Reply:
Adding BATS tests for this behavior.